### DISCUSSION TITLE
How to reference a Property in Files XML element

### DISCUSSION THREAD

**Discussion by siddhsql** [wixtoolset/issues#8819] [Questions]
I am trying to use [this](https://wixtoolset.org/docs/fivefour/#file-harvesting):

```
<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs">
    <Package Name="MyProduct" Version="1.0.0.0" Manufacturer="Example Corporation" UpgradeCode="B0B15C00-1DC4-0374-A1D1-E902240936D5">
        <Files Include="path\to\files\**" />
    </Package>
</Wix>
```

which btw is wrong. `Files` needs to be inside `Feature` element. Anyway what I want to do is this. I have defined a property like this:

```
<Property Id="BIN_DIR" Value="!(wix.BinDir)" Secure="yes" />
```

and I'd like to use this in `Files` like:

```
        <Files Include="[BIN_DIR]\**" />
```

but it does not replace `BIN_DIR` with the value of the property. Is there any way I can get desired result?

**Comment by wmanning** [comment]
A property is only evaluated during MSI run time.  
There are bind variables: $(myvar)  where you use a <?define myvar="someVal"/>.  Those are evaluated before compile.

https://wixtoolset.org/docs/tools/preprocessor/#variables

**Comment by siddhsql** [comment]
>A property is only evaluated during MSI run time

correct - then why is `BIN_DIR` not getting replaced by its value at runtime?

**Reply by robmen** [reply]
You are missing some fundamentals. I'd recommend watching through the [Deployment Dojo S1:E12](https://robmensching.com/deployment-dojo/episodes/s1/e12/all-the-ways-to-change-variables-and-variables-directories-and-properties/) through episode 18.
